### PR TITLE
(doc): Threadsafety for DeviceClient

### PR DESCRIPTION
**Pull request by vinagesh**
## Checklist
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- [ ] This pull-request is submitted against the `master` branch.

## Description of the changes
A customer requested to add this information to our documentation.

### CODE REVIEW

File: iothub/device/src/DeviceClient.cs [hunk 1]
```diff
@@ -14,6 +14,7 @@ namespace Microsoft.Azure.Devices.Client
     /// <summary>
     /// Contains methods that a device can use to send messages to and receive from the service.
     /// </summary>
+    /// <threadsafety static="true" instance="true" />
```
Comment (vinagesh) [inline iothub/device/src/DeviceClient.cs:17]: A customer requested to document this. I will have to watch out for the msdn docs after our next release to see if this works as expected. Also appreciate comments if this is not true. In my understanding, the DeviceClient is thread-safe.

**Review by timtay-microsoft — APPROVED** [review]

**Review by drwill-ms — APPROVED** [review]

**Review by abhipsaMisra — APPROVED** [review]

**Pull request merged**